### PR TITLE
Fix body not being uploaded with unchunked Async.Pipe.

### PR DESCRIPTION
**Pull request by mefyl**
`Body_raw.disable_chunked_encoding` would drain the pipe to get the content-length, but discard the read body, leaving it empty.

### CODE REVIEW

**Comment by mefyl** [comment]
The CI is red but I'm under the impression it's about the setups themselves ? For instance I'm seeing ```#18 19.28 Uncaught exception:
#18 19.28   
#18 19.28   (Sys_error "/etc/localtime: No such file or directory")```

File: cohttp-async/src/client.ml [hunk 1]
```diff
@@ -128,20 +128,19 @@ let callv ?interrupt ?ssl_config uri reqs =
 let call ?interrupt ?ssl_config ?headers ?(chunked=false) ?(body=`Empty) meth uri =
   (* Create a request, then make the request. Figure out an appropriate
      transfer encoding *)
-  let req =
```
Comment (mseri) [inline cohttp-async/src/client.ml:131]: I think it is easier to read if you don't pipe such a large block. I'd suggest
```
let (req, body') = match chunked with
[...]
in
req >>= request ?interrupt ?ssl_config ~body:body' ~uri
```

**Comment by mseri** [comment]
I agree that the CI failures seem all unrelated. If you make the little stylistic change above, I'll be happy to merge

File: cohttp-async/src/client.ml (hunk 1 above)
Comment (mseri) [reply cohttp-async/src/client.ml:131]: I had another look at the code, similar blocks appear in other places. I am going to merge as is

**Pull request merged**